cmake/Modules/CodeCoverage.cmake

```diff
@@ -149,6 +150,9 @@ function(SETUP_TARGET_FOR_COVERAGE_LCOV)
         message(FATAL_ERROR "genhtml not found! Aborting...")
     endif() # NOT GENHTML_PATH
 
+    # Conditional arguments
+    set(GENHTML_ARGS $<$<BOOL:${CPPFILT_PATH}:>:"--demangle-cpp">)
```

Comment:
I think we should _definitely_ do this, as demangling makes the report SO much more readable. It's the difference between this

![image](https://user-images.githubusercontent.com/538020/70380022-e4f34a80-1902-11ea-9ea1-7dc855289dd7.png)


and this

![image](https://user-images.githubusercontent.com/538020/70379965-23d4d080-1902-11ea-8ba7-f981a74941ea.png)

(I mean, it's still not _super_ readable, but...)

Demangling requires the external `c++filt` command, so the addition of this option will be conditional on the `find_program()` at the top of the file locating a working executable.)